@marcrasi There's some strange breakage in parsing, not just `@differentiable`. When I removed those attributes, I'm getting `#tfop` errors from building TensorFlow module.

```
/Users/rxwei/Development/Swift/swift-tf-source-merge/tensorflow-swift-bindings/RawOpsGenerated.swift:13093:10: error: unknown expression syntax exists in the source
  return #tfop("RefSelect",
         ^
/Users/rxwei/Development/Swift/swift-tf-source-merge/tensorflow-swift-bindings/RawOpsGenerated.swift:13118:10: error: unknown expression syntax exists in the source
  return #tfop("RefSwitch",
         ^
```

--- a/lib/Parse/ParseDecl.cpp
+++ b/lib/Parse/ParseDecl.cpp
@@ -1585,7 +1585,7 @@ bool Parser::parseNewDeclAttribute(DeclAttributes &Attributes, SourceLoc AtLoc,
 
   /// SWIFT_ENABLE_TENSORFLOW
   case DAK_Differentiable: {
-    ParserResult<DifferentiableAttr> Attr = parseDifferentiableAttribute(AtLoc, Loc);
+    auto Attr = parseDifferentiableAttribute(AtLoc, Loc);
     if (Attr.isNonNull()) {
       Attributes.add(Attr.get());
     }

--- a/lib/SILOptimizer/Mandatory/TFPartition.cpp
+++ b/lib/SILOptimizer/Mandatory/TFPartition.cpp
@@ -29,6 +29,7 @@
 #include "swift/AST/Expr.h"
 #include "swift/AST/Module.h"
 #include "swift/AST/GenericEnvironment.h"
+#include "swift/AST/GenericSignatureBuilder.h"
 #include "swift/Demangling/Demangle.h"
 #include "llvm/ADT/BitVector.h"
 #include "llvm/ADT/DepthFirstIterator.h"
@@ -90,14 +91,10 @@ static SubstitutionMap getSingleSubstitutionMap(SILFunction *f, Type ty) {
 /// substitution map suitable for calling this builtin.
 static SubstitutionMap getSingleSubstitutionMapForBuiltin(
   Identifier builtinName, Type ty, ASTContext &ctx) {
-  SmallVector<ValueDecl *, 1> builtinDecls;
-  DeclName declName(builtinName);
-  ctx.TheBuiltinModule->lookupMember(builtinDecls, ctx.TheBuiltinModule,
-                                     declName, Identifier());
-  assert(builtinDecls.size() == 1 && "No such builtin!");
-  auto *builtinDecl = cast<FuncDecl>(builtinDecls.front());
-  return SubstitutionMap::get(builtinDecl->getGenericSignature(),
-                              [&](SubstitutableType *t) { return ty; },
+  auto builder = GenericSignatureBuilder(ctx);
+  builder.addGenericParameter(GenericTypeParamType::get(0, 0, ctx));
+  auto *genSig = std::move(builder).computeGenericSignature(SourceLoc());
+  return SubstitutionMap::get(genSig, [&](SubstitutableType *t) { return ty; },
                               MakeAbstractConformanceForGenericType());
 }
 

--- a/stdlib/public/Python/Python.swift
+++ b/stdlib/public/Python/Python.swift
@@ -37,7 +37,7 @@ public typealias OwnedPyObjectPointer = UnsafeMutablePointer<PyObject>
 // - Note: When Swift has ownership, `PyReference` should be removed.
 //   `PythonObject` will define copy constructors, move constructors, etc. to
 //   implement move semantics.
-@_versioned @_fixed_layout
+@usableFromInline @_fixed_layout
 final class PyReference {
   private var pointer: OwnedPyObjectPointer
 
@@ -88,7 +88,7 @@ public struct PythonObject {
   /// The underlying `PyReference`.
   fileprivate var reference: PyReference
 
-  @_versioned
+  @usableFromInline
   init(_ pointer: PyReference) {
     reference = pointer
   }

--- a/stdlib/public/core/FloatingPoint.swift.gyb
+++ b/stdlib/public/core/FloatingPoint.swift.gyb
@@ -1845,7 +1845,7 @@ extension FloatingPoint {
   @inlinable // FIXME(sil-serialize-all)
   @_transparent
   /// SWIFT_ENABLE_TENSORFLOW
-  @differentiable(reverse, withRespectTo: (self), adjoint: _adjointSquareRoot)
+  // @differentiable(reverse, withRespectTo: (self), adjoint: _adjointSquareRoot)
   public func squareRoot( ) -> Self {
     var lhs = self
     lhs.formSquareRoot( )
@@ -1855,7 +1855,7 @@ extension FloatingPoint {
   /// SWIFT_ENABLE_TENSORFLOW
   /// The adjoint of `squareRoot`. Returns the gradient of `squareRoot` with
   /// respect to `self`.
-  @_versioned @_inlineable // FIXME(sil-serialize-all)
+  @inlinable // FIXME(sil-serialize-all)
   func _adjointSquareRoot(originalValue: Self, adjoint: Self) -> Self {
     return 2 * self * adjoint
   }
@@ -1876,9 +1876,9 @@ extension FloatingPoint {
   @inlinable // FIXME(sil-serialize-all)
   @_transparent
   /// SWIFT_ENABLE_TENSORFLOW
-  @differentiable(
-    reverse, withRespectTo: (self, .0, .1), adjoint: _adjointAddingProduct
-  )
+  // @differentiable(
+  //   reverse, withRespectTo: (self, .0, .1), adjoint: _adjointAddingProduct
+  // )
   public func addingProduct(_ lhs: Self, _ rhs: Self) -> Self {
     var addend = self
     addend.addProduct(lhs, rhs)
@@ -1888,7 +1888,7 @@ extension FloatingPoint {
   /// SWIFT_ENABLE_TENSORFLOW
   /// The adjoint of `addingProduct`. Returns the gradient of `addingProduct`
   /// with respect to `self`, `lhs` and `rhs`.
-  @_versioned @_inlineable
+  @inlinable
   func _adjointAddingProduct(
     _ lhs: Self, _ rhs: Self,
     originalValue: Self, adjoint: Self

--- a/stdlib/public/core/FloatingPointTypes.swift.gyb
+++ b/stdlib/public/core/FloatingPointTypes.swift.gyb
@@ -1732,7 +1732,7 @@ extension ${Self} {
   @inlinable // FIXME(sil-serialize-all)
   @_transparent
   /// SWIFT_ENABLE_TENSORFLOW
-  @differentiable(reverse, adjoint: _adjointAdd)
+  // @differentiable(reverse, adjoint: _adjointAdd)
   public static func + (lhs: ${Self}, rhs: ${Self}) -> ${Self} {
     var lhs = lhs
     lhs += rhs
@@ -1742,7 +1742,7 @@ extension ${Self} {
   @inlinable // FIXME(sil-serialize-all)
   @_transparent
   /// SWIFT_ENABLE_TENSORFLOW
-  @differentiable(reverse, adjoint: _adjointSubtract)
+  // @differentiable(reverse, adjoint: _adjointSubtract)
   public static func - (lhs: ${Self}, rhs: ${Self}) -> ${Self} {
     var lhs = lhs
     lhs -= rhs
@@ -1752,7 +1752,7 @@ extension ${Self} {
   @inlinable // FIXME(sil-serialize-all)
   @_transparent
   /// SWIFT_ENABLE_TENSORFLOW
-  @differentiable(reverse, adjoint: _adjointMultiply)
+  // @differentiable(reverse, adjoint: _adjointMultiply)
   public static func * (lhs: ${Self}, rhs: ${Self}) -> ${Self} {
     var lhs = lhs
     lhs *= rhs
@@ -1762,7 +1762,7 @@ extension ${Self} {
   @inlinable // FIXME(sil-serialize-all)
   @_transparent
   /// SWIFT_ENABLE_TENSORFLOW
-  @differentiable(reverse, adjoint: _adjointDivide)
+  // @differentiable(reverse, adjoint: _adjointDivide)
   public static func / (lhs: ${Self}, rhs: ${Self}) -> ${Self} {
     var lhs = lhs
     lhs /= rhs
@@ -1773,7 +1773,7 @@ extension ${Self} {
 /// SWIFT_ENABLE_TENSORFLOW
 /// Adjoints of standard binary operators.
 extension ${Self} {
-  @_versioned @_inlineable // FIXME(sil-serialize-all)
+  @inlinable // FIXME(sil-serialize-all)
   @_transparent
   static func _adjointAdd(
     _ lhs: ${Self}, _ rhs: ${Self}, originalValue: ${Self}, adjoint: ${Self}
@@ -1781,7 +1781,7 @@ extension ${Self} {
     return (adjoint, adjoint)
   }
 
-  @_versioned @_inlineable // FIXME(sil-serialize-all)
+  @inlinable // FIXME(sil-serialize-all)
   @_transparent
   static func _adjointSubtract(
     _ lhs: ${Self}, _ rhs: ${Self}, originalValue: ${Self}, adjoint: ${Self}
@@ -1789,7 +1789,7 @@ extension ${Self} {
     return (adjoint, -adjoint)
   }
 
-  @_versioned @_inlineable // FIXME(sil-serialize-all)
+  @inlinable // FIXME(sil-serialize-all)
   @_transparent
   static func _adjointMultiply(
     _ lhs: ${Self}, _ rhs: ${Self}, originalValue: ${Self}, adjoint: ${Self}
@@ -1797,7 +1797,7 @@ extension ${Self} {
     return (rhs * adjoint, lhs * adjoint)
   }
 
-  @_versioned @_inlineable // FIXME(sil-serialize-all)
+  @inlinable // FIXME(sil-serialize-all)
   @_transparent
   static func _adjointDivide(
     _ lhs: ${Self}, _ rhs: ${Self}, originalValue: ${Self}, adjoint: ${Self}

